src/service/invoice-service.ts

```diff
@@ -500,82 +417,75 @@ export default class InvoiceService {
   }
 
   /**
-   * TEMPORARY: Checks if any of the transactions are already invoiced
-   * This will later be replaced when the InvoiceService is refactored
-   * @param ids
+   * Checks if any of the transactions are already invoiced
+   * @param transactions - The transactions to check
    */
-  public async checkIfSAlreadyInvoiced(ids: number[]) {
-    const transactions = await this.manager.find(Transaction, { where: { id: In(ids) }, relations: ['subTransactions', 'subTransactions.subTransactionRows'] });
+  public async checkIfInvoiced(transactions: Transaction[]) {
     transactions.forEach((t) => {
       t.subTransactions.forEach((tSub) => {
         tSub.subTransactionRows.forEach((tSubRow) => {
-          if (tSubRow.invoice) throw new NotImplementedError('Transaction already invoiced');
+          if (tSubRow.invoice) throw new Error('Transaction already invoiced');
         });
       });
     });
   }
 
-
   /**
-   * Checks if all transaction are for the same seller
-   * @param ids
+   * Gets transactions for an invoice
+   * returns false if the transactions contain invoiced transactions
+   * @param params
    */
-  public async checkSingleSellerTransactions(ids: number[]) {
-    const transactions = await this.manager.find(Transaction, { where: { id: In(ids) }, relations: ['subTransactions', 'subTransactions.to'] });
-    // Get all sellers from sub transactions
-    const sellers = new Set<number>();
+  public async getTransactionsForInvoice(params: InvoiceTransactionRequest): Promise<TransactionResponse[] | false> {
+    const { forId, fromDate, tillDate } = params;
+    const transactionService = new TransactionService(this.manager);
+
+    const tIds = (await transactionService.getTransactions({ fromId: forId, fromDate, tillDate }, {}))
+      .records.map((t) => t.id);
+
+
+    const relations: FindOptionsRelations<Transaction> = {
+      subTransactions: {
+        subTransactionRows: {
+          invoice: true,
+          product: {
+            vat: true,
+          },
+        },
+        container: true,
+      },
+      pointOfSale: true,
+    };
+
+    const transactions = await this.manager.find(Transaction, { where: { id: In(tIds) },
+      relations });
```

Comment:
Is there no way we can do all this with the existing `TransactionService.getTransactions` method? Because this feels very redundant.

Reply:
No, getTransactions returns a paginated base response which does not allow the user to extract invoice information.

The transaction service also needs the getOptions refactor, sadly.